encoders/src/main/scala/org/apache/spark/sql/execution/columnar/SharedExternalStoreUtils.scala

```diff
@@ -0,0 +1,183 @@
+/*
+ *
+ */
+package org.apache.spark.sql.execution.columnar
+
+import java.sql.{Connection, PreparedStatement, SQLException, Types}
+import java.util.Properties
+
+import scala.collection.immutable.HashMap
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+import scala.util.Random
+
+import org.apache.spark.sql.AnalysisException
+import org.apache.spark.sql.execution.ConnectionPool
+import org.apache.spark.sql.row.SnappyStoreClientDialect
+import org.apache.spark.sql.sources.ConnectionProperties
+import org.apache.spark.sql.types.{Decimal, StructType}
+
+object SharedExternalStoreUtils {
+
+  private var useLocatorURL: Boolean = _
+
+  final val COLUMN_BATCH_SIZE = "COLUMN_BATCH_SIZE"
+  final val COLUMN_MAX_DELTA_ROWS = "COLUMN_MAX_DELTA_ROWS"
+
+  private def useLocatorUrl(hostList: ArrayBuffer[(String, String)]): Boolean =
+    hostList.isEmpty
+
+  def getTableSchema(schemaAsJson: String): StructType = StructType.fromString(schemaAsJson)
+
+  def getConnection(connectionProperties: ConnectionProperties,
+      hostList: ArrayBuffer[(String, String)]): Connection = {
+    useLocatorURL = useLocatorUrl(hostList)
+    createConnection(connectionProperties, hostList)
+  }
+
+  private def createConnection(connProperties: ConnectionProperties,
+      hostList: ArrayBuffer[(String, String)]): Connection = {
+    // TODO CHECK THIS SOCKETCREATOR DEPENDENCY RESOLUTION
+    // val localhost = SocketCreator.getLocalHost
```

Comment:
Use ClientSharedUtils.getLocalHost. build currently has a dependency on snappydata-store-core so this should be available. Should reduce the dependency to gemfire-shared and snappydata-store-shared instead of core.

Reply:
Used the ClientSharedUtils.getLocalHost as this dependency was already available here